.github/workflows/php-qa.yml

```diff
@@ -1,77 +1,95 @@
 name: PHP Quality Assurance
 on:
-  push:
-  # Allow manually triggering the workflow.
-  workflow_dispatch:
-
-# Cancels all previous workflow runs for the same branch that have not yet completed.
+    push:
+    pull_request:
+    workflow_dispatch:
 concurrency:
-  # The concurrency group contains the workflow name and the branch name.
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 
 jobs:
-    qa:
+    static-qa:
         runs-on: ubuntu-latest
-        if: "!contains(github.event.head_commit.message, 'ci skip')"
-        strategy:
-            fail-fast: true
-            matrix:
-                php-versions: ['5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1']
-                dependency-versions: ['lowest', 'highest']
+        if: ${{ !contains(github.event.head_commit.message, 'skip qa') }}
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
 
-                include:
-                - php-versions: '8.2'
-                  dependency-versions: 'highest'
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 7.4
+                    coverage: none
+                    tools: cs2pr
 
-        continue-on-error: ${{ matrix.php-versions == '8.2' }}
+            -   name: Install dependencies
+                uses: ramsey/composer-install@v2
 
+            -   name: Check code styles
+                run: |
+                    ./vendor/bin/phpcs -ps . --standard=PHPCompatibility --ignore=*/vendor/* --extensions=php --basepath=./ --report-full --report-checkstyle="phpcs-report.xml" --runtime-set testVersion 5.6- --runtime-set ignore_warnings_on_exit 1
+                    cs2pr --graceful-warnings phpcs-report.xml
+
+    unit-tests:
+        runs-on: ubuntu-latest
+        if: ${{ !contains(github.event.head_commit.message, 'skip tests') }}
+        env:
+            USE_COVERAGE: 'no'
+        strategy:
+            fail-fast: false
```

Comment:
Why are you changing `fail-fast` ?

Reply:
Because when there's failure I wanna see if the problem exists only on that version or it is a "generic" error. With fail-fast all versions fail if one fails, and I don't have the chance to know if it was a version-specifc error.